Add more debugging for readiness test timeout

grcevski opened the pull request:
This failure here https://github.com/elastic/elasticsearch/issues/86664 happened again. It could essentially mean two things, the initial 10 second timeout I chose is not enough or that I have another concurrency bug somewhere. 

With this PR I'm extending the timeout to 30 seconds and I've also put an assert to see what the await status is when we fail. In case the timeout hasn't expired and we still see this error, that would mean it's an unknown concurrency issue of some kind.

Relates #86664

Commit:
Extend and assert on wait timeout

Commit:
Merge branch 'master' into fix_readiness_test_timeout

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

rjernst commented:
> It could essentially mean two things, the initial 10 second timeout I chose is not enough or that I have another concurrency bug somewhere.

The latter seems more likely. 10 seconds should be enough for basically any test? Can we add anymore debugging to help diagnose the next time we see this failure (for example, some additional test logging so we can see how long and when certain parts of this test are running)?

Commit:
Merge branch 'master' into fix_readiness_test_timeout

Commit:
Add more instrumentation

grcevski commented:
@rjernst, I added extra instrumentation and reverted back the timeout to 10 seconds. Given it failed again last night, I think with this extra info I'll be able to at least pinpoint where the issue is.

rjernst reviewed (approved):
LGTM

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.3](https://github.com/elastic/elasticsearch/pull/87762)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/87762">](https://github.com/elastic/elasticsearch/pull/87762) |
